### DISCUSSION TITLE
Image proeprty rotation-angle panic on software_renderer?

### DISCUSSION THREAD

**Discussion by zhangzqs** [slint-ui/slint#5308] [General]
When I used property `rotation-angle` on the Image component, my program panic and the error was reported at:

https://github.com/slint-ui/slint/blob/fbc441c3723b72df5269426237fa4dc2919986f3/internal/core/software_renderer.rs#L2460-L2462

Are there plans to support rotation in the software renderer?

**Comment by ogoffart** [comment]
There is indeed a few things that are not yet implemented in the software renderer.
The current limitation are documented there: https://releases.slint.dev/1.6.0/docs/slint/src/advanced/backends_and_renderers#software-renderer
The rotation of image is tracked there: https://github.com/slint-ui/slint/issues/3068

Long term we want to remove these limitations and bring the software renderer on par with the GPU renderer
